shared/js/background/events/3p-tracking-cookie-blocking.js

```diff
@@ -12,26 +27,19 @@ function dropTracking3pCookiesFromResponse (request) {
     const tab = tabManager.get({ tabId: request.tabId })
     let responseHeaders = request.responseHeaders
 
-    if (tab && tab.site.isFeatureEnabled('trackingCookies3p') && request.type !== 'main_frame') {
-        if (!trackerutils.isTracker(request.url)) {
+    if (tab && request.type !== 'main_frame') {
+        const requestIsTracker = trackerutils.isTracker(request.url)
+        if (!shouldBlockHeaders(request, tab, requestIsTracker)) {
             return { responseHeaders }
         }
 
-        // Strip 3rd party response header
-        if (!request.responseHeaders) return { responseHeaders }
-        if (!tab) {
-            const initiator = request.initiator || request.documentUrl
-            if (!initiator || trackerutils.isFirstPartyByEntity(initiator, request.url)) {
-                return { responseHeaders }
-            }
-        } else if (tab && trackerutils.isFirstPartyByEntity(request.url, tab.url)) {
-            return { responseHeaders }
-        }
-        if (!utils.isCookieExcluded(request.url)) {
+        // Strip 3rd party cookie response header
+        const cookieFeature = requestIsTracker ? 'trackingCookies3p' : 'nonTracking3pCookies'
+        if (!utils.isCookieExcluded(request.url, cookieFeature)) {
```

Comment:
I *think* you can delete the earlier bail out 31-37. (happy for this to be a follow up)

```suggestion
        if (shouldBlockHeaders(request, tab, requestIsTracker)) {
```

Reply:
Same for the other one below also.

Reply:
I filed: https://app.asana.com/0/0/1202339603569796/f feel free to reformat etc